### PR TITLE
feat(pr): only check pr title on title changed

### PR DESCRIPTION
**Pull request by Drarig29**
# What this PR is about?

This PR moves the "Check PR Title" job to a separate file, to have a different trigger configuration.

See [events-that-trigger-workflows#pull-request](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request) for a list of possible trigger event types.
And [webhook-events-and-payloads#pull-request](https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#pull_request) for their documentation.

image

According to this note, the default config (if `types` is omitted) is: `[opened, synchronize, reopened]`.

I think we don't need the `synchronize` and `reopened` events to trigger a PR title check here, since `synchronize` is triggered when the PR's base branch is updated and `reopened` is when a PR is... reopened.

### CODE REVIEW

File: .github/workflows/pull-request-title.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+name: Pull Request Title
+
+on:
+  pull_request:
+    types: [opened, edited]
```
Comment (Drarig29) [inline .github/workflows/pull-request-title.yml:5]: The title will be checked when the PR is opened or edited. However, a PR will also be seen as "edited" if its **body** is updated... and we can't fine tune this behavior.

But, given we cancel in-progress checks, a contributor won't be able to spam too much.

**Comment by Drarig29** [comment]
Works as expected:

**Comment by bdebon** [comment]
This is already an improvment I made yesterday to put the check before the test-and-build. Your solution is even cleaner but I wonder if it's gonna fix my real problem which is: CI will not run automatically for new contributor. I have to manually accept to run the ci for everyone of them which is super painful. Do you think this specific job could be launch for anyone or there is no other way to overcome that in a public repository (I understand the reason they do that...)

**Comment by Drarig29** [comment]
Hi! It's not exactly what you did: you made it first to run.

But once you (@bdebon) approve to run the workflows for a contributor, if the PR title check was failing and the contributor just changes the title to make the CI pass, the job won't re-run.

What my change does is that it will re-run when the approved contributor changes the PR title. It's just a nice to have, I agree.

---

There is no way to override the behavior of approving for a first-time contributor for a single CI job. This is a repository-level setting.

However, if you are not paying for the CI you could just disable this setting and trust contributors, if this is so painful (and I understand).

You can find the option in your repo's settings, Code Automation, Actions, General and here:

**Pull request merged**